test/hotspot/jtreg/compiler/vectorization/TestFloat16ToFloatConv.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+* @test
+* @key randomness
+* @bug 8350835
+* @summary Test bug fix for JDK-8350835 discovered through Template Framework
+* @requires vm.compiler2.enabled
+* @library /test/lib /
+* @run main/othervm -XX:-TieredCompilation -XX:CompileOnly=compiler.vectorization.TestFloat16ToFloatConv::test* compiler.vectorization.TestFloat16ToFloatConv
+*/
+
+package compiler.vectorization;
+
+import compiler.lib.ir_framework.*;
+import jdk.test.lib.Utils;
+import java.util.Random;
+
+public class TestFloat16ToFloatConv {
+    private static final Random RANDOM = Utils.getRandomInstance();
+    private static final int SIZE = 1024;
+    private static byte[] aB = new byte[SIZE];
+    private static short[] aS = new short[SIZE];
+    private static int[] aI = new int[SIZE];
+    private static long[] aL = new long[SIZE];
+    private static float[] goldB, goldS, goldI, goldL;
+
+    static {
+        for (int i = 0; i < aB.length; i++) {
+            aB[i] = (byte)RANDOM.nextInt();
+            aS[i] = (short)RANDOM.nextInt();
+            aI[i] = RANDOM.nextInt();
+            aL[i] = RANDOM.nextLong();
+        }
+        goldB = testByteKernel(aB);
+        goldS = testShortKernel(aS);
+        goldI = testIntKernel(aI);
+        goldL = testLongKernel(aL);
+    }
+
+    @Test
```

Comment:
```suggestion
    @Test
    @IR(failOn = { IRNode.VECTOR_CAST_HF2F }, applyIfCPUFeatureOr = { "avx512vl", "true", "f16c", "true" })
```

Reply:
I left out the IR check because we do intend to vectorize this going forward. Instead the bug fix is verified by checkResult.  Also the fix is not specific to Intel platform so if we do add IR check it will need to be generic. 
@eme64 your thoughts please? Would you like to see an IR check here that vectorization is not happening?

Reply:
Personally, I generally prefer to have `failOn` IR rules, if we expect that at least for now there should be no vectorization. But add a comment why we expect no vectorization, so that if it ever does vectorize and the IR rule fails the person has a hint, and does not have to reverse-engineer too much. And if it turns out that we should one day vectorize, then we already have all these tests ready to just flip the `failOn` into `count`.